### ISSUE TITLE
Trouble interpreting CRC8 results

### ISSUE THREAD

**Issue by funkyluke** [triq-org/revdgst#5]
i am having trouble interpreting the output of revsum for CRC8:
```
Code len is 8 bytes (16 nibbles), 43 codes read.
Row weights:  Max 22 /64 bit (34.4%)  Min 12 /64 bit (18.8%)  Avg 17.0 /64 bit (26.6%)
Processing...
Complete message...
Found: crc8(&b[0], 7, 0x2f, 0x03) ^ b[7] == 0; // (100.0%)
Found: crc8(&b[0], 7, 0xca, 0xe6) ^ b[7] == 0; // (62.8%)
```
so my understanding would be that for 100% of the lines, calculating CRC8 over the first 7 bytes with Poly 0x2F and Init 0x03 results in a CRC8 result of byte 8 of the line - correct?
However, when I calculate the CRC8 using the CRC8 algo in utlis.h, I do not get byte 8 ?!
Am I doing something wrong or is this a bug ?

**Comment by zuckschwerdt** [comment]
Yes, that's what should work. Also it should be `crc8(b, 8, 0x2f, 0x03) == 0`
Though poly 0x2f does not look like a proper CRC, also init here is just too random, we expect something like 0, 0xff, 0xaa, 0xcc, ...

**Comment by funkyluke** [comment]
so here's some of the data i'm using:
```
00 24 c0 89 06 54 4c c8
00 24 c0 89 06 54 4d e7
00 24 c0 89 06 4b 4c a9
00 24 c0 89 06 44 4d c2
00 24 c0 89 06 3c 4d bc
00 24 c0 89 06 37 4c 02
00 24 c0 89 06 31 4d 05
00 24 c0 89 06 2c 4c b6
00 24 c0 89 06 28 4c 63
00 24 c0 89 06 24 4d 1c
00 24 c0 89 06 1a 4c f1
00 24 c0 89 06 17 4c 48
00 24 c0 89 06 14 4c 5c
00 24 c0 89 06 12 4c 74
00 24 c0 89 06 10 4c 89
00 24 c0 89 06 47 50 bc
00 24 c0 89 06 46 50 55
```
I tried to re-implement the crc8 sweep but so far I was not able to find a solution for e.g. the first 3

**Comment by funkyluke** [comment]
> Yes, that's what should work. Also it should be `crc8(b, 8, 0x2f, 0x03) == 0` Though poly 0x2f does not look like a proper CRC, also init here is just too random, we expect something like 0, 0xff, 0xaa, 0xcc, ...

no, this check also fails, result is 0xdd. I would have expected some standard value as well but you never know with proprietary stuff...

**Comment by zuckschwerdt** [comment]
It works, but init is a nicer 0xaa. I have to check what's wrong with the output.

**Comment by zuckschwerdt** [comment]
Ah ok, the number given as init is really the fix point when using an init of 0x00. I would have to run that "backwards" through the CRC to get an actual init.

**Comment by zuckschwerdt** [comment]
Fixed by 57b9762
